### PR TITLE
Python support update

### LINKED ISSUE #13: Update supported Python versions
- drop support for Python versions < 3.6
- add support for Python versions 3.7, 3.8 (if missing)
- fix linting errors (if present)

If you are a beginner, and need to help for this issue, especially during Hacktoberfest, please comment here or reach out to me via https://jugmac00.github.io/

### PR DESCRIPTION
**Pull request by sgaist**
This merge request updates the minimal Python version to 3.7 and adds 3.8.

It also adds the tools required to run the tests to list of dependencies.

Fixes #13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
+python: 3.7
 env:
-    - TOXENV=py27
-    - TOXENV=py33
-    - TOXENV=py34
-    - TOXENV=py35
-    - TOXENV=pypy
+    - TOXENV=py37
+    - TOXENV=py38
     - TOXENV=pypy3
```
Comment (jugmac00) [inline .travis.yml:7]: Could you please update above code to be similar to https://github.com/morepath/morepath_cerebral_todomvc/commit/c5b413728e05e92423c162534c32e6b39ee1cb0f - looks like Travis does no longer install several python versions on its boxes, and so we need to specify the needed versions in a `matrix`.

File: setup.py [hunk 2]
```diff
@@ -22,12 +22,8 @@
...
         "License :: OSI Approved :: BSD License",
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.7",
-        "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.3",
-        "Programming Language :: Python :: 3.4",
-        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
     ],
```
Comment (jugmac00) [inline setup.py:27]: Please add Python 3.6

File: setup.py [hunk 3]
```diff
@@ -38,6 +34,8 @@
         test=[
             'pytest-cov',
             'WebTest >= 2.0.14'
+            'tox',
+            'flake8'
         ],
```
Comment (jugmac00) [inline setup.py:39]: As mentioned in the other repos - flake8 should not be necessary, as it gets installed via tox (deps=...).

File: tox.ini [hunk 4]
```diff
@@ -1,15 +1,16 @@
 [tox]
-envlist = py27,py33,py34,py35,pypy,pypy3,pep8
+envlist = py37,py38,pypy3,pep8
```
Comment (jugmac00) [inline tox.ini:3]: please add python 3.6

File: tox.ini [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 
 [testenv]
 deps = -e{toxinidir}[test]
-commands = py.test {posargs}
+usedevelop = True
```
Comment (jugmac00) [inline tox.ini:6]: Was the `usedevelop = True` necessary to run the tests? If so - can you explain the reasoning behind? I am curious.

**Review by jugmac00 — CHANGES_REQUESTED** [review]
LGTM with some comments - please try to fix them.

**Comment by jugmac00** [comment]
Please, could you also add a change log entry?

**Comment by jugmac00** [comment]
I copied over the text over from https://github.com/morepath/more.webassets/pull/6 as the here we have the same issues.

Hi @sgaist

currently, CI is broken, see https://travis-ci.org/github/morepath/more.webassets/builds/731987458

The reason seems that Travis nowadays need the "matrix" syntax for configuring several Python interpreters, also see here
morepath/morepath_cerebral_todomvc@c5b4137

As a side note .. I just saw you force-pushed some new changes. While this works, there is one problem with this. Above I added some inline annotations on what needs to be done for this pr - as you force pushed a new state, these do not match any more. On the other hand, if you just added another commit, I could easily see if you applied the requested changes. Do not worry about too many commits, as the maintainer always can squash all changes into one commit on merging.

So, the rule of thumb I follow - when I work together with other people, especially via GitHub, I do not force push.

Thanks for contributing!

**Comment by sgaist** [comment]
As written on https://github.com/morepath/more.webassets/pull/6, I see your point and I'll do that :-)

File: .travis.yml (hunk 1 above)
Comment (sgaist) [reply .travis.yml]: Done

**Review by jugmac00 — APPROVED** [review]

**Comment by jugmac00** [comment]
LGTM!

There is one minor thing - Travis does no longer test `pypy3`. I guess I sidetracked you when I linked to the issue of the other project which does not use pypy3.

I never used pypy myself and I do not know whether anybody uses it, but as long as the there is no maintenance burden, I'd like to keep it.

I will do a quick PR after this - nothing to worry.

If you regularly contribute to Python projects, it may be a good idea to install `check-python-versions` which is a nice cli tool, which compares the listed Python versions in `tox.ini`, `travis.yml`, `setup.py` and afaik also the gh action config file.

Thanks for your contribution!

File: tox.ini (hunk 5 above)
Comment (sgaist) [reply tox.ini:6]: Yes it was and it looks like it fixed #12.

Honestly, I did not had the time yet to completely understand why it behaved like that in this package while not in others. 

I have taken a look at several other package from the project and saw that in morepath/reg> use it. I understand what it does but I am not completely sure why it is needed here yet.

**Comment by sgaist** [comment]
Thanks for the trick ! I did not know about it.

If everything goes fine with pypy3 then I agree with you, it's better to keep it. More coverage is good.

By the way, I saw that you are adding black to others of your project, would you like to do it for this one as well ?

**Comment by jugmac00** [comment]
While the `morepath` framework is still used by e.g. the German pirate party, there is currently low traction. I consider it as in maintenance mode. For me that means keep it working, e.g. regularly add support for new Python versions, fix deprecation warnings and so on.

Personally, I am interested in this project in order to learn the differences how web frameworks work.

The creator of this framework wrote a blog post which I highly recommend:
https://blog.startifact.com/posts/framework-patterns/

That said - I do not plan to canonicalize the repositories, or introduce new tooling like e.g. `pre-commit` which I usually use for the other projects I maintain. I certainly would review a pull request.

**Comment by henri-hulski** [comment]
I recently black to the project in agreement with Martijn.
I also removed Python 2 for the part of the project that I maintain. 
Also integrated black with pre-commit in these packages. 
If you want to continue to add black to other packages that would be great.

**Comment by sgaist** [comment]
Done in #16

**Pull request merged**